aws/resource_aws_elasticsearch_domain.go

```diff
@@ -212,7 +212,7 @@ func resourceAwsElasticSearchDomain() *schema.Resource {
 						"instance_type": {
 							Type:     schema.TypeString,
 							Optional: true,
-							Default:  "m3.medium.elasticsearch",
+							Default:  elasticsearch.ESPartitionInstanceTypeM3MediumElasticsearch,
```

Comment:
This is a nice catch but I am slightly confused why this is defaulted anyway and why to this specific instance type when I can't see any mention of it in the AWS docs at all.

Unfortunately removing this default would be a breaking change :crying_cat_face: